src/components/tools/table-tool/use-controls-column.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback } from "react";
+import AddColumnSvg from "../../../assets/icons/add/add.nosvgo.svg";
+import RemoveRowSvg from "../../../assets/icons/remove/remove.nosvgo.svg";
+import { TFormatterProps } from "./grid-types";
+
+interface IUseControlsColumn {
+  readOnly?: boolean;
+  onAddColumn?: () => void;
+  onRemoveRow?: (rowId: string) => void;
+}
+export const useControlsColumn = ({ readOnly, onAddColumn, onRemoveRow }: IUseControlsColumn) => {
+
+  const ControlsHeaderRenderer: React.FC = useCallback(() => {
+    return !readOnly
+            ? <AddColumnButton onAddColumn={onAddColumn} />
+            : null;
+  }, [onAddColumn, readOnly]);
+
+  const ControlsRowFormatter: React.FC<TFormatterProps> = useCallback(({ row, isRowSelected }) => {
+    return !readOnly && isRowSelected
+            ? <RemoveRowButton rowId={row.__id__} onRemoveRow={onRemoveRow} />
+            : null;
+  }, [onRemoveRow, readOnly]);
+  ControlsRowFormatter.displayName = "ControlsRowFormatter";
+
+  return { ControlsHeaderRenderer, ControlsRowFormatter };
+};
+
+interface IAddColumnButtonProps {
+  onAddColumn?: () => void;
+}
+const AddColumnButton: React.FC<IAddColumnButtonProps> = ({ onAddColumn }) => {
+  return (
+    <div className="add-column-button" onClick={() => onAddColumn?.()}>
+      <AddColumnSvg className="add-column-icon"/>
+    </div>
```

Comment:
I'm seeing some cropping in the Add Column icon.  Do you see this?
![image](https://user-images.githubusercontent.com/5126913/99576504-ec362b80-298e-11eb-9ac7-44919e451609.png)

Reply:
I don't see that. Honestly, it seems like a rendering bug given that the size of the element is sized precisely to the SVG, but I don't think it will hurt anything to either increase the size of the element and/or add `overflow: visible`.

Reply:
Actually, I think there's something more serious going on here. The double circle appearance should only occur on `:active`, i.e. when the mouse is down, in which case you shouldn't have the background color you have on the add column button. How did you capture that image?

Reply:
I was thinking the same. I tinkered with the container and the SVG and looked at the SVGs by themselves.  Everything seems fine.  My only concern is that it appears in multiple browsers.  I see it in both Chrome and Firefox (a little scary is that I can't get CLUE to load in Edge).

Reply:
That's just a simple screenshot with a row selected and the mouse hovering over the add row button (cursor is removed in screenshot).

Reply:
Here's a gif of hovering over the 2 buttons and clicking on the add column.
![svgs](https://user-images.githubusercontent.com/5126913/99578259-71224480-2991-11eb-8c48-406dfa3e7676.gif)